rollup.config.js

```diff
@@ -1,45 +1,72 @@
-import babel from '@rollup/plugin-babel';
+import { getBabelOutputPlugin } from '@rollup/plugin-babel';
 const commonjs = require('@rollup/plugin-commonjs');
 const { nodeResolve } = require('@rollup/plugin-node-resolve');
 const nodePolyfills = require('rollup-plugin-node-polyfills');
+import typescript from '@rollup/plugin-typescript';
 
 // Entry point of the library
-const input = './lib/bson.js';
+const input = './src/bson.ts';
 
 export default [
+  {
+    input,
+    /* Browser ESM bundle */
+    output: {
+      file: 'dist/bson.browser.esm.js',
+      format: 'esm',
+      exports: 'named',
+      sourcemap: true
+    },
+    plugins: [
+      typescript({ tsconfig: './tsconfig.rollup.json' }),
+      nodeResolve({ preferBuiltins: false }),
+      commonjs(),
+      nodePolyfills(),
+      getBabelOutputPlugin({
+        presets: [
+          [
+            '@babel/preset-env',
+            {
+              modules: false,
+              targets: { ie: 8, node: 6 }
+            }
+          ]
+        ]
+      })
+    ]
+  },
   {
     input,
     output: [
-      /* Browser ESM bundle */
-      {
-        file: 'dist/bson.browser.esm.js',
-        format: 'es'
-      },
       /* Browser UMD bundle */
       {
         file: 'dist/bson.browser.umd.js',
-        format: 'umd',
-        name: 'BSON'
+        format: 'esm',
+        name: 'BSON',
+        sourcemap: true
       },
-      /* Browser IIFE bundle */
-      {
+      /* Browser IIFE bundle */ {
         file: 'dist/bson.bundle.js',
-        format: 'iife',
-        name: 'BSON'
+        format: 'esm',
```

Comment:
```suggestion
        format: 'iife',
```

Reply:
The format here is correct, babel is downgrading the code after rollup has done its transformations so babel is handling the bundle format. I use UMD in the babel preset because it is equivalent to an iife (babel doesn't have an iife output without a plugin, which seems unnecessary).